src/common/pageUtils.ts

```diff
@@ -55,7 +58,12 @@ export async function sendRequest(
         }
     );
 
-    if (isSuccessful(response)) throw new Error(errorMessage);
+    if (!response || response.isError || !isSuccessful(response)) {
+        const errorToReport = response.isError ? `The server responded with a status of ${response.status} (${response.statusText}). URL: ${response.url}` :
+            errorMessage;
+        reportError(errorToReport);
+        throw new UserError(errorMessage);
```

Comment:
This could be done internally by your error report logic, for example add an optional parameter to your `UserError`: 
```ts
class UserError extends Error {
    report: boolean;
    constructor(message: string, report = false) {
        super(message);
        this.name = USER_ERROR;
        this.report = report;
    }
}
```

and later in the function `displayError` verify that field:
```ts
    if (
        !isDevelopment() &&
        (error.name !== USER_ERROR || (error as UserError).report)
    )
        Sentry.captureException(error);
```

Reply:
But not all errors are UserErrors. The other errors can be caused by other factors.

Reply:
Correct! this is only for the mentioned example (in the function `sendRequest`) where we throw `UserError` and want that error to be reported.

Reply:
The other problem is that the messages are different.

Reply:
Oh, the idea is that we can define the `UserError` to our needs.

I think a better suitable solution in this case would be to provide a context for the error:
```ts
class UserError extends Error {
    report: boolean;
    context?: CaptureContext;
    constructor(message: string, report = false, context?: CaptureContext) {
        super(message);
        this.name = USER_ERROR;
        this.report = report;
        this.context = context;
    }
}
```
```ts
// in displayError
 if (
        !isDevelopment() &&
        (error.name !== USER_ERROR || (error as UserError).report)
    )
        Sentry.captureException(error, error.context);
```

Reply:
But the concept of the UserError is not like that. I mean we can do this for sure, but UserError is some kind of error that user make and it shoudnt be reported. The reason why i used the UserError in there because maybe we should display a more understandable error. Normally, this error shoudnt be a user error. Instead of this maybe we can delete the user error from the `sendRequest` and just throw an Error.

Reply:
I'm cool with that! I thought that in future we may have other cases where we want to report the user errors, if it's a special case I don't mind leaving it as it is :+1: